coreneuron/utils/reports/report_handler.cpp

```diff
@@ -0,0 +1,198 @@
+#include "report_handler.h"
+#include "coreneuron/utils/reports/nrnsection_mapping.h"
+#include "coreneuron/nrnoc/mech_mapping.hpp"
+
+namespace coreneuron {
+
+void ReportHandler::create_report(double dt, double tstop, double delay) {
+#ifdef ENABLE_REPORTING
+    if (m_report_config.start < t) {
+        m_report_config.start = t;
+    }
+    m_report_config.stop = std::min(m_report_config.stop, tstop);
+
+    m_report_config.mech_id = nrn_get_mechtype(m_report_config.mech_name);
+    if (m_report_config.type == SynapseReport && m_report_config.mech_id == -1) {
+        std::cerr << "[ERROR] mechanism to report: " << m_report_config.mech_name
+                  << " is not mapped in this simulation, cannot report on it" << std::endl;
```

Comment:
Do not unnecessarily flush the stream.
```suggestion
                  << " is not mapped in this simulation, cannot report on it\n";
```